src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/dto/TestResultsDTO.java

```diff
@@ -77,6 +81,43 @@ public ZonedDateTime getRunDate() {
         return runDate;
     }
 
+    @Override
+    public ZonedDateTime getBuildRunDate() {
+        return getRunDate();
+    }
+
+    @Override
+    public Optional<String> getCommitHashFromAssignmentRepo() {
+        final var testRepoNameSuffix = RepositoryType.TESTS.getName();
+        final var firstCommit = getCommits().stream().filter(commit -> !commit.getRepositorySlug().endsWith(testRepoNameSuffix)).findFirst();
+        return firstCommit.map(CommitDTO::getHash);
+    }
+
+    @Override
+    public Optional<String> getCommitHashFromTestsRepo() {
+        final var testRepoNameSuffix = RepositoryType.TESTS.getName();
+        final var firstCommit = getCommits().stream().filter(commit -> commit.getRepositorySlug().endsWith(testRepoNameSuffix)).findFirst();
+        return firstCommit.map(CommitDTO::getHash);
+    }
+
+    @Override
+    public boolean isBuildSuccessful() {
+        final var testSum = getSkipped() + getFailures() + getErrors() + getSuccessful();
```

Comment:
What would happen here if a test is marked with `@Ignore` in Junit (the instructor disabled a test because it does not work as intended, but it is not removed from the test-repo)?
I would assume that this counts as `skipped` test -> The build would not be successful.

Reply:
I'm honestly not sure. I did not modify the production code, I just moved it into this function. I'd guess it's out of scope of this pr?